packages/@ember/-internals/metal/lib/mixin.ts

```diff
@@ -575,6 +576,9 @@ export default class Mixin {
     this.ownerConstructor = undefined;
     this._without = undefined;
 
+    // Eagerly add INIT_FACTORY to avoid issues in DEBUG as a result of Object.seal(mixin)
+    Object.defineProperty(this, INIT_FACTORY, { value: null, writable: true });
```

Comment:
If this is only needed in `DEBUG` builds, can you wrap this in a guard (or move it just below into the existing `if (DEBUG) {` check just below)?

Reply:
Also, why do we need `Object.defineProperty` here? Wouldn't this do the trick:

```js
this[INIT_FACTORY] = null;
```